clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AWSServerGroupNameResolver.groovy

```diff
@@ -45,6 +52,55 @@ class AWSServerGroupNameResolver extends AbstractServerGroupNameResolver {
 
   @Override
   List<AbstractServerGroupNameResolver.TakenSlot> getTakenSlots(String clusterName) {
-    return asgService.getTakenSlots(clusterName)
```

Comment:
`asgService.getTakenSlots` fetched all server groups in order to determine what was available/taken ... switching to using cached cluster data.